File: src/main/java/net/ehicks/bts/beans/Attachment.java
//package net.ehicks.bts.beans;
//
//import net.ehicks.eoi.EOI;
//
//import javax.persistence.*;
//import java.io.Serializable;
//import java.util.Arrays;
//import java.util.Date;
//import java.util.List;
//
//@Entity
//@Table(name = "attachments")
//public class Attachment implements Serializable
//{
//    @Id
//    @GeneratedValue(strategy=GenerationType.AUTO)
//    private Long id;
//
//    @Column(nullable = false)
//    private Long issueId;
//
//    @Column(nullable = false)
//    private Long dbFileId;
//
//    public Attachment()
//    {
//    }
//
//    public Attachment(Long issueId, Long dbFileId)
//    {
//        this.issueId = issueId;
//        this.dbFileId = dbFileId;
//    }
//
//    @Override
//    public boolean equals(Object obj)
//    {
//        if (!(obj instanceof Attachment)) return false;
//        Attachment that = (Attachment) obj;
//        return this.id.equals(that.getId());
//    }
//
//    @Override
//    public int hashCode()
//    {
//        return 17 * 37 * id.intValue();
//    }
//
//    public String toString()
//    {
//        return this.getClass().getSimpleName() + ":" + id.toString();
//    }
//
//    public static List<Attachment> getAll()
//    {
//        return EOI.executeQuery("select * from attachments");
//    }
//
//    public DBFile getDbFile()
//    {
//        return DBFile.getById(dbFileId);
//    }
//
//    public Issue getIssue()
//    {
//        return Issue.getById(issueId);
//    }
//
//    // -------- Getters / Setters ----------
//
//    public Long getId()
//    {
//        return id;
//    }
//
//    public void setId(Long id)
//    {
//        this.id = id;
//    }
//
//    public Long getIssueId()
//    {
//        return issueId;
//    }
//
//    public void setIssueId(Long issueId)
//    {
//        this.issueId = issueId;
//    }
//
//    public Long getDbFileId()
//    {
//        return dbFileId;
//    }
//
//    public void setDbFileId(Long dbFileId)
//    {
//        this.dbFileId = dbFileId;
//    }
//}

File: src/main/java/net/ehicks/bts/beans/Attachment.kt
package net.ehicks.bts.beans

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.time.LocalDateTime
import javax.persistence.*

@Entity
data class Attachment(
        @Id @GeneratedValue(strategy = GenerationType.AUTO)
        val id: Long = 0,

        var name: String = "",
        @ManyToOne(fetch = FetchType.LAZY) val issue: Issue,
        @ManyToOne(fetch = FetchType.LAZY) val dbFile: DBFile,
        @CreatedDate
        val createdOn: LocalDateTime,
        @ManyToOne(fetch = FetchType.LAZY) val createdBy: User?
) {
    fun getShortName(): String {
        return if (name.length > 32) name.substring(0, 26) else name
    }
}

@Repository
interface AttachmentRepository : JpaRepository<Attachment, Long> {
    fun findByIssue_id(issueId: Long): List<Attachment>
    fun findByDbFile_id(dbFileId: Long): List<Attachment>
}